File: src/main/java/moe/sota/decompiler/types/ClassType.java
package moe.sota.decompiler.types;

import moe.sota.decompiler.models.FileModel;
import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
import org.jetbrains.annotations.NotNull;

public class ClassType extends Type {

    public ClassType() {
        super("icons/class.png", SyntaxConstants.SYNTAX_STYLE_JAVA);
    }

    @Override
    public boolean isFormat(@NotNull FileModel fileModel) {
        return fileModel.getName().toLowerCase().endsWith(".class");
    }

}

File: src/main/kotlin/moe/sota/decompiler/types/ClassType.kt
package moe.sota.decompiler.types

import moe.sota.decompiler.models.FileModel
import org.fife.ui.rsyntaxtextarea.SyntaxConstants

object ClassType : Type("icons/class.png", SyntaxConstants.SYNTAX_STYLE_JAVA) {
    override fun isFormat(fileModel: FileModel): Boolean =
        fileModel.name.endsWith(".class", ignoreCase = true)
}
